pennylane/optimize/qng.py

```diff
@@ -192,13 +192,17 @@ def step_and_cost(
 
         if recompute_tensor or self.metric_tensor is None:
             if metric_tensor_fn is None:
-                metric_tensor_fn = qml.metric_tensor(qnode, diag_approx=self.diag_approx)
+
+                if self.diag_approx:
+                    metric_tensor_fn = qml.metric_tensor(qnode, approx="diag")
+                else:
+                    metric_tensor_fn = qml.metric_tensor(qnode)
```

Comment:
@antalszava do you think we should update the QNG optimizer to match the change in the metric tensor?

That is,

```python
class QNGOptimizer:

    def __init__(self, stepsize=0.01, approx="diag", lam=0):
```

?

I had forgotten the QNG optimizer also used this old terminology! I think it makes sense to match the metric tensor here, that way we can simply have

```suggestion
                metric_tensor_fn = qml.metric_tensor(qnode, approx=self.approx)
```